src/lib/request.ts

```diff
@@ -53,9 +28,55 @@ type PromisifiedRequest = typeof requestLib & {
 	) => Bluebird<any>;
 };
 
-const requestHandle = requestLib.defaults(exports.requestOpts);
+const getRequestInstances = once(async () => {
+	// Generate the user agents with out versions
+	const osVersion = await osRelease.getOSVersion(constants.hostOSVersionPath);
+	const osVariant = await osRelease.getOSVariant(constants.hostOSVersionPath);
+	let userAgent = `Supervisor/${supervisorVersion}`;
+	if (osVersion != null) {
+		if (osVariant != null) {
+			userAgent += ` (Linux; ${osVersion}; ${osVariant})`;
+		} else {
+			userAgent += ` (Linux; ${osVersion})`;
+		}
+	}
+
+	const requestOpts: requestLib.CoreOptions = {
+		gzip: true,
+		timeout: DEFAULT_REQUEST_TIMEOUT,
+		headers: {
+			'User-Agent': userAgent,
+		},
+	};
+
+	const resumableOpts = {
+		timeout: DEFAULT_REQUEST_TIMEOUT,
+		maxRetries: DEFAULT_REQUEST_RETRY_COUNT,
+		retryInterval: DEFAULT_REQUEST_RETRY_INTERVAL,
+	};
+
+	const requestHandle = requestLib.defaults(exports.requestOpts);
```

Comment:
I don't see `requestOpts` being exported in this file.

Reply:
good catch!